registry/remote/repository_test.go

```diff
@@ -2210,6 +2210,98 @@ func Test_BlobStore_Resolve(t *testing.T) {
 	}
 }
 
+func Test_Manifest_generateManifestDescriptorWithVariousDockerContentDigestHeaders(t *testing.T) {
+	type testIOStruct struct {
+		name                    string
+		clientSuppliedReference string
+		serverCalculatedDigest  digest.Digest // for non-HEAD (body-containing) requests only
+		errExpectedOnHEAD       bool
+		errExpectedOnGET        bool
+	}
+	const theAmazingBanClan = "Ban Gu, Ban Chao, Ban Zhao"
+	const theAmazingBanDigest = "b526a4f2be963a2f9b0990c001255669eab8a254ab1a6e3f84f1820212ac7078"
+	correctDigest := fmt.Sprintf("sha256:%v", theAmazingBanDigest)
+	incorrectDigest := fmt.Sprintf("sha256:%v", "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
```

Comment:
There is no concept of `incorrectDigest`. As long as the digest is in a right form, it is valid although it may not be desired. In that case, it is a customer error.

Reply:
I'm having problems convincing myself of that; could you help me understand how?  I've got the two code-paths traversed [here](https://github.com/oras-project/oras-go/pull/237#discussion_r954289594).